### PR TITLE
Use thnn version of Tanh/Sigmoid instead of autograd.

### PR DESCRIPTION
**Pull request by csarofeen**
Will help with using nn.functional.sigmoid and nn.functional.tanh by reducing number of backwards kernels.

### CODE REVIEW

**Comment by jekbradbury** [comment]
This should slightly reduce the Python method-call overhead here, but I'm confused as to how it would change the number of kernels executed?

**Comment by csarofeen** [comment]
Currently https://github.com/pytorch/pytorch/blob/master/torch/autograd/_functions/pointwise.py#L57-L70 is being called, which is 4 kernels in backwards.

**Review by soumith — APPROVED** [review]

**Comment by jekbradbury** [comment]
Wait, why is there a separate non-THNN implementation in the first place? Now `torch.tanh` will have the same semantics but worse performance than `F.tanh`

**Comment by Smerity** [comment]
Just wanted to thank @csarofeen for finding this. While I absolutely adore the cuDNN accelerated RNN cells, there are times when I need to implement something custom and don't have the time / flexibility to optimize at the CUDA level. Improving the runtime speed of custom non-fused non-cuDNN RNN cells has a direct impact on my research :)

**Comment by ngimel** [comment]
@jekbradbury, yes, now F.tanh backward performance would be better than torch.tanh, but before @csarofeen's PR the backward pass would call 4 kernels instead of 1.

**Comment by csarofeen** [comment]
@Smerity https://github.com/pytorch/pytorch/pull/1119 would also help if you ever find yourself using lstm cell/gru cell without cudnn.
